advanced/security.md

```diff
@@ -2,46 +2,141 @@
-description: A statement on security.
-keywords: security xss scripting vulnerability hack hacker
+description_short: Security information for TinyMCE.
+description: Information on reporting security issues, what TinyMCE does to protect users, and what you can do to protect your users.
+keywords: security xss scripting vulnerability hack hacker csp mitigation protection protect
 ---
 
-## Q: Is TinyMCE protected against XSS vulnerabilities?
+> **Note**: The following is _general_ security advice that may be relevant to website or application using {{site.productname}}.
 
-{{site.productname}} filters out some of the more common XSS content like scripts from the content since it is common that the editor is used in a single page application. For additional security, consider passing it through server-side filters like [HTMLPurifier](http://htmlpurifier.org/).
+## Overview
 
-## Q: How do I setup Content Security Policy (CSP) with TinyMCE?
+- [Reporting TinyMCE security issues](#reportingtinymcesecurityissues)
+- [What we do to maintain TinyMCEs' security](#whatwedotomaintaintinymcessecurity)
+
+  - [Scripts and XSS vulnerabilities](#scriptsandxssvulnerabilities)
+  - [Keeping dependencies up-to-date](#keepingdependenciesup-to-date)
+  - [Configuring Content Security Policy (CSP) for TinyMCE](#configuringcontentsecuritypolicycspfortinymce)
+- [Potential security risks and mitigation measures](#potentialsecurityrisksandmitigationmeasures)
+
+  - [Cross-Site Scripting (XSS)](#cross-sitescriptingxss)
+  - [Injection](#injection)
+  - [Insecure Transmission and Storage of data](#insecuretransmissionandstorageofdata)
+  - [Broken Authentication and Session Management](#brokenauthenticationandsessionmanagement)
+  - [Using Components with Known Vulnerabilities](#usingcomponentswithknownvulnerabilities)
+  - [Further security information](#furthersecurityinformation)
+
+## Reporting TinyMCE security issues
+
+{{site.companyname}} values the work of security researchers in improving the security of technology products worldwide. We welcome researchers who wish to responsibly disclose vulnerabilities in our products or systems. Note that we do not offer any “bug bounty” program or any form of payment for disclosed vulnerabilities. If you would like to report a vulnerability, please email `infosec@tiny.cloud`.
+
+## What we do to maintain TinyMCEs' security
+
+To protect {{site.productname}}s' users, {{site.companyname}}:
+
+- Patches Cross-Site Scripting (XSS) vulnerabilities,
+- Keeps {{site.productname}}s' dependencies up to date, and
+- Provides information how to configure a Content Security Policy that works with {{site.productname}}.
+
+### Scripts and XSS vulnerabilities
+
+{{site.productname}} filters out most Cross-Site Scripting (XSS) content such as scripts from the editor content. For additional security, {{site.companyname}} recommends passing the editor content through server-side filters such as [HTMLPurifier](http://htmlpurifier.org/).
```

Comment:
This isn't really for additional security, it's more so that any client-side only security can be circumvented, so they must do server-side filtering to ensure their published content remains safe.

Reply:
Yeah - sanitizing content server-side is essential. I'm not sure I would recommend a specific tool. I wouldn't say we filter out "most" XSS content - just say "we filter content such as scripts"

Reply:
updated to:
> TinyMCE filters content such as scripts from the editor content, however, client-side applications can be by-passed by attackers. Tiny recommends processing received editor content through server-side filters.